### PR TITLE
Update workflows and actions

### PR DESCRIPTION
**Pull request by Nicoretti**
- Add manual workflow trigger for building and deploying GH pages
- Simplify workflows by creating an action for  python & poetry setup
- Update changelog fragment template
- Bump version number

### CODE REVIEW

File: changelog.d/20230315_084229_nicola.coretti.rst [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+.. - A bullet item for the Fixed category.
+..
+.. 🔐 Security
+.. -----------
+..
+.. - A bullet item for the Security category.
+..
+🧰 Internal
+-----------
+- Simplify workflows by facotring out python & poetry setup into an action
```
Comment (ckunki) [inline changelog.d/20230315_084229_nicola.coretti.rst]:
```suggestion
- Simplify workflows by factoring out python & poetry setup into an action
```

File: changelog.d/templates/fragment-template.rst [hunk 2]
```diff
@@ -32,3 +32,8 @@
 ..
 .. - A bullet item for the Security category.
 ..
+.. 🧰 Internal
+.. -----------
+..
+.. - A bullet item for the Security category.
```
Comment (ckunki) [inline changelog.d/templates/fragment-template.rst]:
```suggestion
.. - A bullet item for the Internal category.
```

File: .github/workflows/gh-pages.yaml [hunk 3]
```diff
@@ -5,30 +5,25 @@ on:
       - master
     tags:
       - "*"
+    workflow_dispatch:
```
Comment (ckunki) [inline .github/workflows/gh-pages.yaml:8]: Is this intentionally blank?

**Review by ckunki — CHANGES_REQUESTED** [review]

File: .github/workflows/gh-pages.yaml (hunk 3 above)
Comment (Nicoretti) [reply .github/workflows/gh-pages.yaml:8]: Yes

**Review by ckunki — APPROVED** [review]

**Pull request merged**